### ISSUE TITLE
How to test another dataset

### ISSUE THREAD

**Issue by iamheejae** [htzheng/ECCV2018_CrossNet_RefSR#4]
I trained the flower dataset with your network and now i want to test with other dataset which contains different flowers. 
I found that your code load each train and test dataset but i cant' find where the  loaded test dataset is used. I want to test my image with the checkpoint file that i trained. 
Furthermore, if i want to see the result HR image where can i found the result image? I just fix the code in "train_multi_warping.py" like below, but i can't assure that this is correct way to see the HR output image. 

            for i in range(pre_npy.shape[0]):
                psnr_ += psnr(pre_npy[i],label_img_npy[i]) / pre_npy.shape[0]
		                
		img_bgr = pre_npy[i].transpose(1,2,0)
		img_rgb = cv2.cvtColor(img_bgr, cv2.COLOR_BGR2RGB)
		#cv2.imshow('result', img_rgb)
		imgplot = plt.imshow(img_rgb)
		plt.show()
		#cv2.imwrite('./result/image_'+str(i)+'.png', img_rgb)
		#cv2.waitKey(500)
           	print (i,psnr(pre_npy[i],label_img_npy[i]))

I also try to change "train_multi_warping.py" as shown below to test image, but error comes out that "too many indices for tensor of dimension 4". 

            buff_val = dataset_test.nextBatch_new(batchsize=config['batch_size'], shuffle=True, view_mode = 'Random', augmentation = False, offset_augmentation=config['data_displacement_augmentation'], crop_shape = config['train_data_crop_shape'])
            
            val_img1_LR = buff_val['input_img1_LR']
            val_img2_HR = buff_val['input_img2_HR']

            val_img = np.concatenate((val_img1_LR,val_img2_HR),axis = 1)
            val_label_img = buff['input_img1_HR']

            val_img = torch.from_numpy(val_img)
            val_img = val_img.cuda()
            with torch.no_grad():
                val_pred = net(val_img) 
                val_pred_npy = val_pred.cpu().numpy()
                psnr_ = 0
                for i in range(val_pred_npy.shape[0]):
                    psnr_ += psnr(val_pred_npy[i],val_label_img[i]) / val_pred_npy.shape[0]
                    print (i,psnr(val_pred_npy[i],val_label_img[i]))

**Comment by BKZero** [comment]
excuse me, but where you get the Flower Dataset? I also want to use my own dataset, but firstly, i want to run the flower dataset example. But i did not see any link or readme about the dataset.

**Comment by iamheejae** [comment]
you can download the dataset from the link below.

https://github.com/pratulsrinivasan/Local_Light_Field_Synthesis

also, the paper information(which offer flower dataset) is mentioned in crossnet paper, as reference number [47]. 

thanks.

​
-----Original Message-----

**Comment by cherry5219** [comment]
excuse me, I want to ask if I only need lr and Hr pictures when training. Do I need to get optical flow pictures as gt? If not, how can I ensure that the trained flownet can correctly predict optical flow? I have just touched this field and some parts of the article and code are not very understandable。
